cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetectorImpl.java

```diff
@@ -63,8 +64,7 @@
 
   // Subject
 
-  private Subject<FailureDetectorEvent, FailureDetectorEvent> subject =
-      PublishSubject.<FailureDetectorEvent>create().toSerialized();
+  private DirectProcessor<FailureDetectorEvent> subject = DirectProcessor.create();
```

Comment:
Use `DirectProcessor.create().serialize()` and add `FluxSink<FailureDetectorEvent> sink = subject.sink()`

Reply:
replace `subject.onCompleted();` at line 171 to `sink.complete()`

Reply:
replace `subject.onNext(new FailureDetectorEvent(member, status));` at line 369 with `sink.next(...)`